### PR TITLE
Reverse a string #1999

### PR DESCRIPTION
**Pull request by Amit366**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

I have made a program for reversing a string and also updated the readme.md file.

### CODE REVIEW

**Comment by atarax665** [comment]
Couldn't find any issue related to this PR.

**Comment by Amit366** [comment]
Sir it is the part of implementation in Java  #1999

**Comment by Amit366** [comment]
And also I was assigned.

**Comment by Amit366** [comment]
It was my fault that I didn't mention above but I mentioned the issue number #1999

File: Java/cp/ReverseString.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        // Reversing the string by extracting each element from the end of the string and storing in a variable result
+
+        for (int i = str.length()-1; i >= 0; i--)
+            result+= str.charAt(i);
+
+        // Printing the reversed string
+        
+        System.out.println(result);
+    }
+}
```
Comment (atarax665) [inline Java/cp/ReverseString.java]: Add Sample Input/Output with space and time complexity of the program in commented form.

File: Java/cp/ReverseString.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        // Accepting input from user
+
+        System.out.println("Enter string");
+        String str = sc.nextLine();
+        String result="";
+          
+        // Reversing the string by extracting each element from the end of the string and storing in a variable result
+
+        for (int i = str.length()-1; i >= 0; i--)
+            result+= str.charAt(i);
```
Comment (atarax665) [inline Java/cp/ReverseString.java]:
```suggestion
            result += str.charAt(i);
```

File: Java/cp/ReverseString.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+        // Accepting input from user
+
+        System.out.println("Enter string");
+        String str = sc.nextLine();
+        String result="";
+          
+        // Reversing the string by extracting each element from the end of the string and storing in a variable result
+
+        for (int i = str.length()-1; i >= 0; i--)
```
Comment (atarax665) [inline Java/cp/ReverseString.java]:
```suggestion
        for (int i = str.length() - 1; i >= 0; i--)
```

File: Java/cp/ReverseString.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+class ReverseString {
+    public static void main(String[] args)
+    {
+        Scanner sc = new Scanner(System.in);
+
+        // Accepting input from user
+
+        System.out.println("Enter string");
+        String str = sc.nextLine();
+        String result="";
```
Comment (atarax665) [inline Java/cp/ReverseString.java]:
```suggestion
        String result = "";
```

**Review by atarax665 — CHANGES_REQUESTED** [review]

**Comment by Amit366** [comment]
Sir I have updated the suggested changes.

File: Java/cp/ReverseString.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
+import java.lang.*;
```
Comment (HarshCasper) [inline Java/cp/ReverseString.java]: Add a beginner pitch about your Code

File: Java/cp/ReverseString.java [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+/* Time Complexity is O(n)
+   Space Complexity is O(n)
+   
+   Input :
+   Hello World
+   
+   Output :
+   dlroW olleH
+*/
```
Comment (HarshCasper) [inline Java/cp/ReverseString.java]: Add a new line at the EOF

File: Java/cp/ReverseString.java [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+        // Accepting input from user
+
+        System.out.println("Enter string");
+        String str = sc.nextLine();
+        String result = "";
+          
+        // Reversing the string by extracting each element from the end of the string and storing in a variable result
+
+        for (int i = str.length() - 1; i >= 0; i--)
+            result += str.charAt(i);
```
Comment (HarshCasper) [inline Java/cp/ReverseString.java]: Push your main Code Logic to a different function. Use the main() function only as your driver code

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: Java/cp/ReverseString.java (hunk 6 above)
Comment (Amit366) [reply Java/cp/ReverseString.java]: Sir could you elaborate a bit

**Review by vatsalkesarwani12 — APPROVED** [review]

**Review by atarax665 — APPROVED** [review]

**Comment by Amit366** [comment]
Sir please add the level and DWOC tags so that it is counted

**Comment by Amit366** [comment]
@atarax665 please add the DWOC and level tags so that it is considered for points.

**Pull request merged**